packages/sku/src/utils/polyfillDetector.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  POLYFILL_REGISTRY,
+  type PolyfillRegistryEntry,
+} from './polyfillRegistry.js';
+
+export interface DetectedPolyfill extends PolyfillRegistryEntry {
+  polyfillName: string;
+}
+
+/**
+ * Detects unnecessary polyfills from the configured polyfills array
+ * by matching package names against the registry.
+ */
+export const detectUnnecessaryPolyfills = (
+  polyfills: string[],
+): DetectedPolyfill[] =>
+  polyfills.flatMap((polyfillName) => {
+    const registryEntry = POLYFILL_REGISTRY[polyfillName];
+
+    if (!registryEntry) {
+      return [];
+    }
+
+    return [
+      {
+        polyfillName,
+        ...registryEntry,
+      },
+    ];
+  });
```

Comment:
`nit`: Performance isn't an issue here so I recommend filtering and mapping separately. It's a little simpler and easier to  comprehend compared to returning empty arrays and flattening them out

```suggestion
export const detectUnnecessaryPolyfills = (
  polyfills: string[]
): DetectedPolyfill[] =>
  polyfills
    .filter((polyfillName) => polyfillName in POLYFILL_REGISTRY)
    .map((polyfillName) => ({
      polyfillName,
      ...POLYFILL_REGISTRY[polyfillName]!,
    }));
```